src/openforms/formio/typing/base.py

```diff
@@ -44,7 +45,13 @@ class OpenFormsConfig(TypedDict):
     translations: NotRequired[TranslationsDict]
     components: NotRequired[AddressValidationComponents]
     requireVerification: NotRequired[bool]
-    dataSrc: NotRequired[Literal["manual", "variable", "referentielijsten"]]
+    dataSrc: NotRequired[
+        Literal[
+            DataSrcOptions.manual,
+            DataSrcOptions.variable,
+            DataSrcOptions.referentielijsten,
+        ]
+    ]
```

Comment:
I'm fairly certain type checkers don't understand this - type checking is done statically and no python code is actually evaluated.

![image](https://github.com/user-attachments/assets/05f38f84-f6ba-48f6-883e-9922e7598a26)


```suggestion
    dataSrc: NotRequired[DataSrcOptions]
```

using the choices class should help enforce that comparisons are made with the constant rather than magic strings, but I haven't tested this.